`xmlsec`-related failures in CI/CD

### Description

Example run: https://github.com/rucio/rucio/actions/runs/8731861204/job/23957949882

```
Building wheel for xmlsec (pyproject.toml): finished with status 'error'
...
#10 41.68       /tmp/pip-install-fquw_0ma/xmlsec_4147c5a4238940d59fb35621ed7f959e/src/keys.c:218:65: error: ‘xmlSecKeyDataFormatEngine’ undeclared (first use in this function); did you mean ‘xmlSecKeyDataFormat’?
#10 41.68         218 |         key->handle = xmlSecCryptoAppKeyLoad(engine_and_key_id, xmlSecKeyDataFormatEngine, NULL, xmlSecCryptoAppGetDefaultPwdCallback(), (void*)engine_and_key_id);
#10 41.68             |                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~
#10 41.68             |                                                                 xmlSecKeyDataFormat
#10 41.68       /tmp/pip-install-fquw_0ma/xmlsec_4147c5a4238940d59fb35621ed7f959e/src/keys.c:218:65: note: each undeclared identifier is reported only once for each function it appears in
#10 41.68       error: command '/usr/bin/gcc' failed with exit code 1
#10 41.68       [end of output]
```

This started occurring between 2024-04-17 and 2024-04-18, tests were passing fine prior to that: https://github.com/rucio/rucio/actions

I'm fairly sure this is not due to a change on our side, it's likely an issue with `xmlsec` itself. 

### Steps to reproduce

N/A

### Rucio Version

_No response_

### Additional Information

_No response_

--- a/requirements.txt
+++ b/requirements.txt
@@ -49,6 +49,7 @@ pycodestyle==2.11.0                                         # New package replac
 pylint==3.0.2
 astroid==3.0.1
 virtualenv==20.24.7                                         # Virtual Python Environment builder
+xmlsec==1.3.13                                              # Required to install pyproject.toml-based projects
 xmltodict==0.13.0                                           # Makes working with XML feel like you are working with JSON
 pytz==2023.3.post1                                          # World timezone definitions, modern and historical
 pydoc-markdown~=4.8.2                                       # Used for generating Markdown documentation for docusaurus

